app/models/concerns/attribute_blockable.rb

```diff
@@ -0,0 +1,372 @@
+  #
+  # @private
+  class BlockedAttributesPreloader < Module
+    # @param method_names [Array<Symbol, String>] Attribute names to preload
+    def initialize(*method_names)
+      @method_names = Array.wrap(method_names).map(&:to_s)
+      super()
+    end
+
+    # Called when this module extends a relation. Sets up preloading hooks.
+    #
+    # @param relation [ActiveRecord::Relation] The relation being extended
+    # @return [ActiveRecord::Relation] The extended relation
+    def extended(relation)
+      add_method_to_preload_list(relation)
+      override_exec_queries(relation)
+      define_preloading_methods(relation)
+      relation
+    end
+
+    private
+      # Adds method names to the relation's preload list, merging with any
+      # existing methods from previous +with_blocked_attributes_for+ calls.
+      def add_method_to_preload_list(relation)
+        existing_methods = relation.instance_variable_get(:@_blocked_attributes_methods) || Set.new
+        relation.instance_variable_set(:@_blocked_attributes_methods, Set.new(existing_methods + @method_names))
+      end
+
+      # Overrides the relation's +exec_queries+ method to trigger preloading
+      # after records are fetched from the database.
+      def override_exec_queries(relation)
+        relation.define_singleton_method(:exec_queries) do |&block|
+          @records = super(&block)
+          preload_blocked_attributes! unless relation.instance_variable_get(:@_blocked_attributes_preloaded)
+          @records
+        end
+      end
+
+      # Defines the preloading methods on the relation instance.
+      def define_preloading_methods(relation)
+        # Iterates through all registered method names and preloads their blocked status
+        relation.define_singleton_method(:preload_blocked_attributes!) do
+          return if @records.blank?
+
+          (@_blocked_attributes_methods || Set.new).each do |method_name|
+            preload_blocked_attribute_for_method(method_name)
+          end
+
+          relation.instance_variable_set(:@_blocked_attributes_preloaded, true)
+        end
+
+        # Preloads blocked status for a single attribute across all records
+        # in the relation using a single MongoDB query.
+        #
+        # @param method_name [String] The attribute method name
+        relation.define_singleton_method(:preload_blocked_attribute_for_method) do |method_name|
+          values = @records.filter_map { |record| record.try(method_name).presence }.uniq
+          return if values.empty?
+
+          # Look up the actual attribute type from the blockable_attributes registry
+          # For example: form_email maps to :email, form_email_domain maps to :email_domain
+          model_class = @records.first.class
+          blockable_config = model_class.blockable_attributes.find { |attr| attr[:blockable_method] == method_name.to_sym }
+          attribute_type = blockable_config ? blockable_config[:attribute] : method_name.to_sym
```

Comment:
Will `blockable_config` have `:attribute` key at this point? only `blockable_method` and `object_type`, right?